networking_ccloud/ml2/agent/eos/switch.py

```diff
@@ -32,6 +31,121 @@
+        except AttributeError:
+            # probably never connected (no channel present)
+            pass
+        # FIXME: currently this throws Except (not retriable)
+        #   would possibly raise a ConnectionRefusedError
+        #   ConnectionRefusedError: [Errno 111] Connection refused
+        LOG.debug("Connecting to switch %s", self._switch_name)
+        try:
+            self._gnmi.connect(*args, **kwargs)
+        except Exception as e:
+            raise cc_exc.SwitchConnectionError(f"{self._switch_name} connect() {e.__class__.__name__} {e}")
+
+    def _run_method(self, method, *args, retries=3, **kwargs):
+        try:
+            start_time = time.time()
+            data = getattr(self._gnmi, method)(*args, **kwargs)
+            LOG.debug("Command %s() succeeded in %.2fs", method, time.time() - start_time)
+            return data
+        except Exception as e:
+            LOG.exception("Command %s() failed on %s in %.2fs (retries left: %s): %s (%s)",
+                          method, self._switch_name, time.time() - start_time, retries, e.__class__.__name__, str(e))
+
+            cmd = f"{method}("
+            if args:
+                cmd += ", ".join(map(json.dumps, args))
+            if kwargs:
+                if args:
+                    cmd += " "
+                cmd += ", ".join(f"{k}={json.dumps(v)}" for k, v in kwargs.items())
+            cmd += ")"
+            LOG.debug("Failed command on %s was %s", self._switch_name, cmd)
+            if isinstance(e, AttributeError):
+                # module 'grpc' has no attribute '_channel', sometimes a first-connect problem
+                LOG.info("Reconnecting %s because of %s %s", self._switch_name, e.__class__.__name__, str(e))
+                self.connect()
+            if retries > 0:
+                time.sleep(0.5)
+                return self._run_method(method, *args, retries=retries - 1, **kwargs)
+            raise cc_exc.SwitchConnectionError(f"{self._switch_name} {method}() {e.__class__.__name__} {e}")
+
+    def get(self, *args, unpack=True, single=True, **kwargs):
+        data = self._run_method("get", *args, **kwargs)
+        if data and unpack:
+            data = data['notification']
+            if single:
+                data = data[0]['update'][0]['val']
+        return data
+
+    def set(self, *args, retries=3, **kwargs):
+        return self._run_method("set", *args, **kwargs)
+        # if isinstance(ret, grpc._channel._InactiveRpcError):
+        #     if ret.code() in (grpc.StatusCode.UNAVAILABLE, grpc.StatusCode.FAILED_PRECONDITION) \
+        #             and retries > 0:
+        #         time.sleep(0.5)
+        #         return self.set(*args, retries=retries - 1, **kwargs)
+        #     raise ret
+        # return ret
+
+
+class EOSSetConfig:
+    def __init__(self):
+        self.delete = []
+        self.replace = []
+        self.update = []
+        self.update_cli = []
+
+    def get_list(self, op):
```

Comment:
Need comment to tell me why only update and replace would be supported as `.delete` and `update_cli` are lists, too ... e.g. "list is not meant as python list but as $something"

Reply:
As we discussed this somewhat off-github... The method is just a helper to avoid a lot of `if op == ...` and update/replace have a similar structure, while delete looks completely different. If you have a better name for this I'd add it, but from the method I don't feel like it really warrants a comment, as it becomes pretty clear when looking at the rest of the code. I can add one if you feel strongly about it improving readability of the code for someone not that into this driver.

Reply:
I don't feel strongly about it, as I'm now more into the code and the usage of this function :D